File: 6-fault-injection/middleware.js
#!/usr/bin/env node

process.stdin.resume();
process.stdin.setEncoding('utf8');
process.stdin.on('data', function(data) {
  var parsed_json = JSON.parse(data);
  parsed_json.response.status = 200;
  parsed_json.response.body = JSON.stringify([
    {
      "origin": "Mars",
      "destination": "Pluto",
      "cost": "9999.99",
      "when": "00:00"
    }
  ]);

  process.stdout.write(JSON.stringify(parsed_json));
});

File: 6-fault-injection/middleware.py
#!/usr/bin/env python

import sys
import json
import logging
import random

logging.basicConfig(filename='middleware.log', level=logging.DEBUG)
logging.debug('Middleware "modify_request" called')

def main():
    payload = sys.stdin.readlines()[0]

    logging.debug(payload)

    payload_dict = json.loads(payload)
    payload_dict['response']['status'] = random.choice([200, 201])

    if "response" in payload_dict and "body" in payload_dict["response"]:
        payload_dict["response"]["body"] = "[{\"origin\": \"Mars\",\"destination\": \"Pluto\",\"cost\": \"9999.99\",\"when\": \"00:00\"}\n]"

    print(json.dumps(payload_dict))

if __name__ == "__main__":
    main()
